libs/@local/graph/postgres-store/src/store/postgres/mod.rs

```diff
@@ -1178,37 +1192,44 @@ where
                     WHERE actor_role.actor_id = $1
                 ),
                 -- We filter out policies that don't apply to the actor's type or principal ID
-                policy AS (
+                policy_edition AS (
                     -- global and actor-type based policies
-                    SELECT policy.*
-                    FROM policy
-                    WHERE principal_id IS NULL
+                    SELECT policy_edition.*
+                    FROM policy_edition
+                    WHERE transaction_time @> now()
+                      AND principal_id IS NULL
                       AND (actor_type IS NULL OR actor_type = $2)
 
                     UNION ALL
 
                     -- actor type policies
-                    SELECT policy.*
-                    FROM policy
+                    SELECT policy_edition.*
+                    FROM policy_edition
                     JOIN principals
-                      ON policy.principal_id = principals.id
-                     AND policy.principal_type = principals.principal_type
-                    WHERE policy.actor_type IS NULL OR policy.actor_type = $2
+                      ON policy_edition.principal_id = principals.id
+                     AND policy_edition.principal_type = principals.principal_type
+                    WHERE policy_edition.transaction_time @> now()
+                      AND (policy_edition.actor_type IS NULL OR policy_edition.actor_type = $2)
                 )
                 SELECT
-                    policy.id,
-                    policy.name,
-                    policy.effect,
-                    policy.principal_id,
-                    policy.principal_type,
-                    policy.actor_type,
-                    policy.resource_constraint,
+                    policy_edition.id,
+                    policy_edition.name,
+                    policy_edition.effect,
+                    policy_edition.principal_id,
+                    policy_edition.principal_type,
+                    policy_edition.actor_type,
+                    policy_edition.resource_constraint,
                     array_remove(array_agg(policy_action.action_name), NULL)
-                FROM policy
-                LEFT JOIN policy_action ON policy.id = policy_action.policy_id
+                FROM policy_edition
+                LEFT JOIN policy_action ON policy_edition.id = policy_action.policy_id
```

Comment:
don't we need a where clause on the transaction time as well here?

Reply:
Yes, thank you!
```suggestion
                LEFT JOIN policy_action
                       ON policy_edition.transaction_time @> now()
                      AND policy_edition.id = policy_action.policy_id
```